uint16 motor_num
uint16 ref
